CONTRIBUTING.md

```diff
@@ -0,0 +1,187 @@
+- bash
+- gcc
+- m4
+- autoconf
+- automake
+- make
+- bison
+- gettext
+- autopoint
+- help2man
+- tar
+- gzip
+- lzip
+- texinfo
+- texlive
+- indent
+- flex (flex bootstraps itself from its previous version)
+
+Configuring your Flex development environment to match the maintenance baseline will help you collaborate and review contributions with the other developers working on Flex.
+
+## Branching Convention
+
+Flex source code is maintained in Git at GitHub. Each Flex developer forks their own copy of the main repository at [westes/flex](https://github.com/westes/flex). Flex development occurs in feature branches of the forks. PRs will eventually be submitted from the feature branches of the forks to westes/flex. 
+
+Flex development forks are not required to reside on GitHub, merely on a publicly-accessible Git server. However:
+- GitHub's pull request (PR) process only works with repositories hosted on GitHub;
+- Flex's automated testing pipeline depends on GitHub's Actions infrastructure and may not function on other Git servers.
+
+NB: If you cannot use GitHub, another developer can help you import your changes to a GitHub name space using multiple remotes. The details are beyond the scope of this manual.
+
+## Maintaining the Test Suite
+
+### Introduction to Flex Tests
+Flex includes an extensive test suite in the `tests/` directory. It includes functional tests as well as regression tests. If you add a feature to Flex, develop corresponding tests for the correct behaviors. Whenever you develop a bug fix, develop a corresponding test for the incorrect behavior to expose future regressions.
+
+The test suite is controlled by `tests/Makefile.am`. Each test consists of a Flex scanner specification, `<category>.l`, that exercises the behavior under test as specifically as possible and an input text file, `<category>.txt`, containing strings for the test scanner to recognize. Flex tests return 0 (zero) on success and 1 (or non-zero) on failure. Test return values are generated by a main() function defined in the test scanner.
+
+In some cases, the test scanner files are generated from a `<category>.rules` file. The file names of generated test scanners are annotated with the combination of conditions they test, like `<category>_nr.l` and `<category>_r.l` for 'non re-entrant' and 're-entrant' versions of a test scanner. Any test scanner is permitted to include annotations in its file name to clarify the conditions that it tests. For example `<category>_c99.l` indicates the test scanner emits C99 actions and should only be used with the C skeleton.
+
+Functional tests, like those testing the combinations of options and their values, are typically implemented in the `<category>.rules` files. Make rules that generate and execute tests from the `<category>.rules` files are found toward the end of `tests/Makefile.am`. Regression tests (and few complex functional tests) are  typically implemented in the `<category>.l` files. 
+
```

Comment:
It's not so much that one can trace back to check_programs as that is what the test harness uses to know what "make check" does. Probably worth making that connection explicit.